google/dev/delete_resources.py

```diff
@@ -0,0 +1,399 @@
+#!/usr/bin/python
+#
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=import-error
+# pylint: disable=star-args
+"""
+Implements program for deleting Google API resources by their name and/or age.
+
+Example usage:
+  # Delete all images in my project more than 30 days old.
+  python delete_resources.py \
+      compute.images \
+      --project=myproject \
+      --age=30 \
+      --dry_run
+
+  # Delete all the versions of all the objects in gs://mybucket
+  python delete_resources.py \
+      storage.objects \
+      --delete_kwargs=bucket=mybucket \
+      --list_kwargs=versions=true \
+      --dry_run
+
+  # Delete all the instances from this project that have 'xyz' in the name.
+  # Use the list_aggregated method, which considers all zones.
+  python delete_resources.py \
+      compute.instances \
+      --project= \
+      --aggregated \
+      --name=".*xyz.*" \
+      --dry_run
+
+  # Delete all the instances from this project in the 'us-central1-f' zone
+  # that have 'xyz' in the name.
+  python delete_resources.py \
+      compute.instances \
+      --project= \
+      --delete_kwargs=zone=us-central1-f \
+      --name=".*xyz.*" \
+      --dry_run
+"""
+
+
+import argparse
+import datetime
+import os
+import re
+import sys
+
+import apiclient
+import httplib2
+import urllib2
+
+from oauth2client.client import GoogleCredentials
+from oauth2client.service_account import ServiceAccountCredentials
+
+
+PLATFORM_FULL_SCOPE = 'https://www.googleapis.com/auth/cloud-platform'
+
+
+def get_metadata(relative_url):
+  """Return metadata value.
+
+  Args:
+    relative_url: [string] Metadata url to fetch relative to base metadata URL.
+  """
+  base_url = 'http://metadata/computeMetadata/v1/'
+  url = os.path.join(base_url, relative_url)
```

Comment:
Probably want `urljoin` instead of `os.path.join` here: [urljoin](https://docs.python.org/2/library/urlparse.html#urlparse.urljoin). I don't know if anyone would execute this script on windows, but it wouldn't work.